packages/fuels-contract/src/contract.rs

```diff
@@ -410,17 +413,116 @@ where
     }
 
     /// Create a CallResponse from call receipts
-    pub fn get_response(&self, receipts: Vec<Receipt>) -> Result<CallResponse<D>, Error> {
+    pub fn get_response(&self, mut receipts: Vec<Receipt>) -> Result<CallResponse<D>, Error> {
         match self.contract_call.output_param.as_ref() {
             None => Ok(CallResponse::new(D::from_token(Token::Unit)?, receipts)),
             Some(param_type) => {
-                let (token, receipts) = ContractCall::get_decoded_output(param_type, receipts)?;
+                let token = ContractCall::get_decoded_output(param_type, &mut receipts)?;
                 Ok(CallResponse::new(D::from_token(token)?, receipts))
             }
         }
     }
 }
 
+#[derive(Debug)]
+#[must_use = "contract calls do nothing unless you `call` them"]
+/// Helper that handles bundling multiple calls into a single transaction
+pub struct MultiContractCallHandler {
+    pub contract_calls: Option<Vec<ContractCall>>,
+    pub tx_parameters: TxParameters,
+    pub wallet: LocalWallet,
+    pub fuel_client: FuelClient,
+}
+
+impl MultiContractCallHandler {
+    pub fn new(wallet: LocalWallet) -> Self {
+        Self {
+            contract_calls: None,
+            tx_parameters: TxParameters::default(),
+            fuel_client: wallet.get_provider().unwrap().client.clone(),
+            wallet,
+        }
+    }
+
+    /// Adds a contract call to be bundled in the transaction
+    /// Note that this is a builder method
+    pub fn add_call<D: Tokenizable>(&mut self, call_handler: ContractCallHandler<D>) -> &mut Self {
+        match self.contract_calls.as_mut() {
+            Some(c) => c.push(call_handler.contract_call),
+            None => self.contract_calls = Some(vec![call_handler.contract_call]),
+        }
+        self
+    }
+
+    /// Sets the transaction parameters for a given transaction.
+    /// Note that this is a builder method
+    pub fn tx_params(&mut self, params: TxParameters) -> &mut Self {
+        self.tx_parameters = params;
+        self
+    }
+
+    /// Returns the script that executes the contract calls
+    pub async fn get_script(&self) -> Script {
+        Script::from_contract_calls(
+            self.contract_calls.as_ref().unwrap().iter().collect(),
+            &self.tx_parameters,
+            &self.wallet,
+        )
+        .await
+    }
+
+    /// Call contract methods on the node, in a state-modifying manner.
+    pub async fn call<D: Tokenizable + Debug>(&self) -> Result<CallResponse<D>, Error> {
+        Self::call_or_simulate(self, false).await
```

Comment:
At first I liked this but after changing it, I got second thoughts. Boolean flags that alter behavior are usually a bad sign. But IMO, not duplicating the 3 lines is useful because of the debug log and the functional-metal-jacket obfuscates stuff.

Reply:
I'm ok with being pragmatic just as long as we're making an informed decision :)  We can always revisit if we find every other fn starting to have a 'doThisIfEnabledThatIfDisabled' flag